lib/fencing/st_output.c

```diff
@@ -76,33 +76,52 @@ stonith__history_description(stonith_history_t *history, bool full_history,
     GString *str = g_string_sized_new(256); // Generous starting size
     char *retval = NULL;
 
-    g_string_printf(str, "%s of %s %s",
-                    stonith_action_str(history->action), history->target,
-                    state_str(history));
+    g_string_printf(str, "%s of %s",
+                    stonith_action_str(history->action), history->target);
+
+    if (!show_details) { // More human-friendly
+        if (((history->state == st_failed) || (history->state == st_done))
+            && (history->delegate != NULL)) {
+            g_string_append_printf(str, " by %s", history->delegate);
+        }
+        g_string_append_printf(str, " for %s@%s",
+                               history->client, history->origin);
+    }
+
+    g_string_append_printf(str, " %s", state_str(history));
 
     // For failed actions, add exit reason if available
     if ((history->state == st_failed) && (history->exit_reason != NULL)) {
         g_string_append_printf(str, " (%s)", history->exit_reason);
     }
 
-    g_string_append(str, ": ");
+    if (show_details) { // More technical
+        g_string_append(str, ": ");
 
-    // For completed actions, add delegate if available
-    if (((history->state == st_failed) || (history->state == st_done))
-        && (history->delegate != NULL)) {
-        g_string_append_printf(str, "delegate=%s, ", history->delegate);
-    }
+        // For completed actions, add delegate if available
+        if (((history->state == st_failed) || (history->state == st_done))
+            && (history->delegate != NULL)) {
 
-    // Add information about originator
-    g_string_append_printf(str, "client=%s, origin=%s",
-                           history->client, history->origin);
+            g_string_append_printf(str, "delegate=%s, ", history->delegate);
+        }
+
+        // Add information about originator
+        g_string_append_printf(str, "client=%s, origin=%s",
+                               history->client, history->origin);
 
-    // For completed actions, add completion time
-    if ((history->state == st_failed) || (history->state == st_done)) {
+        // For completed actions, add completion time
+        if ((history->state == st_failed) || (history->state == st_done)) {
+            char *completed_time = time_t_string(history->completed);
+
+            g_string_append(str,
+                            (full_history? ", completed" : ", last-failed"));
+            g_string_append_printf(str, "='%s'", completed_time);
+            free(completed_time);
+        }
+    } else { // More human-friendly
         char *completed_time = time_t_string(history->completed);
 
-        g_string_append(str, (full_history? ", completed" : ", last-failed"));
-        g_string_append_printf(str, "='%s'", completed_time);
+        g_string_append_printf(str, " at %s", completed_time);
         free(completed_time);
```

Comment:
what was it about the time-stamp in the pending case.
iirc it was when it started - up to now we didn't print it - so that it rather might be 'since' in that case.

Reply:
We currently don't track the initiated time, actually -- that's T458 :)

We only show the time here if the action completed

Reply:
hmm ... what prevents us from adding the - as you say invalid (wasn't sure if we had taken care of that yet) - timestamp in the human-friendly & pending case?
in the more technical case I get how we don't print a timestamp with pending actions but how are we getting the 'last-successful' there?

Reply:
whoops, I thought that was guarded by "if complete", will update

Reply:
updated top commit